torch/tensor.py

```diff
@@ -593,7 +593,7 @@ def __iter__(self):
             warnings.warn('Iterating over a tensor might cause the trace to be incorrect. '
                           'Passing a tensor of different shape won\'t change the number of '
                           'iterations executed (and might lead to errors or silently give '
-                          'incorrect results).', category=RuntimeWarning)
+                          'incorrect results).', category=torch.jit.TracerWarning, stacklevel=2)
```

Comment:
Why should this one be a TracerWarning?  That doesn't look related to jit tracing at all.

Reply:
The warning only issues during tracing, and is about how the trace will not generalize to a different shape.